knowledge-base/telerik-reporting-r3-2016---upgrade-issues-with-objectdatasource-component.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Telerik Reporting R3 2016 - Upgrade Issues with ObjectDataSource Component
+description: Upgrade error with the ObjectDataSource Component after upgrading to R3 2016 or newer.
+type: troubleshooting
+page_title: The ObjectDataSource Component does not work after upgrading to R3 2016
+slug: telerik-reporting-r3-2016---upgrade-issues-with-objectdatasource-component
+res_type: kb
+---
+
+## Environment
+<table>
+	<tr>
+		<td>Product</td>
+		<td>Progress® Telerik® Reporting</td>
+	</tr>
+    <tr>
+		<td>Version</td>
+		<td>R3 2016+</td>
+	</tr>
+	<tr>
+		<td>Report Item</td>
+		<td>ObjectDataSource Component</td>
+	</tr>
+</table>
+
+## Error Message
+
+```
+An exception has occurred while processing 'REPORTNAME' item:
+
+System.Security.SecurityException: The assembly "EXTERNAL_ASSEMBLY_TYPE" is not permitted to be used by an ObjectDataSource component. Please include it in an AssemblyReferences element in the Telerik.Reporting configuration section of your application configuration file.
+```
+
+## Possible Causes
+
+- The assembly supposed to be used by the ObjectDataSource is not referenced in the application configuration file;
+
+- The assembly is incorrectly referenced;
+  
+## Description
+
+As of [R3 2016](../upgrade-path-2016-r3#api-breaking-changes) we provide stricter rules for external assemblies used by reports to get data. This is a breaking change that may lead to the following error on preview, even after running the [Upgrade Wizard](../ui-upgrade-wizard). 
+
+## Solution
+
+ The ObjectDataSource component is designed to read the assembly information from the current project's configuration file.  
+  
+ The settings should be applied in:  
```

Comment:
Which settings? I don't understand how to resolve the issue at all with this guidance.